### PR TITLE
package: update flit and py-pyelftools versions

### PR DESCRIPTION
**Pull request by mikebentley15**

### CODE REVIEW

File: var/spack/repos/builtin/packages/flit/package.py [hunk 1]
```diff
@@ -3,27 +3,32 @@
...
     detecting variability in floating-point code caused by variations in
     compiler code generation, hardware and execution environments."""
 
     homepage = "https://pruners.github.io/flit"
     url      = "https://github.com/PRUNERS/FLiT"
-    url      = "https://github.com/PRUNERS/FLiT/archive/v2.0-alpha.1.tar.gz"
+    url      = "https://github.com/PRUNERS/FLiT/archive/v2.1.0.tar.gz"
 
+    version('2.1.0', sha256='b31ffa02fda1ab0f5555acdc6edc353d93d53ae8ef85e099f83bcf1c83e70885')
     version('2.0-alpha.1', sha256='8de2bd400acf0f513d69f3dbf588e8984dfb18b8ccaaf684391811a0582f694b')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/flit/package.py:18]: You can probably remove this alpha release now that a stable release version is out.

File: var/spack/repos/builtin/packages/flit/package.py [hunk 2]
```diff
@@ -3,27 +3,32 @@
 #
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
 
-
 from spack import *
 
-
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/flit/package.py:9]: Removing this line is causing Flake8 to fail

File: var/spack/repos/builtin/packages/flit/package.py [hunk 3]
```diff
@@ -3,27 +3,32 @@
...
-    depends_on('py-toml', type='run')
-
-    @property
-    def install_targets(self):
-        return ['install', 'PREFIX=%s' % self.prefix]
+    depends_on('python@3:',      type='run')
+    depends_on('py-toml',        type='run')
+    depends_on('py-pyelftools',  type='run')
+
+    #depends_on('bash',           type='run')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/flit/package.py:27]: This line is causing flake8 to fail, remove it if you don't need it

File: var/spack/repos/builtin/packages/flit/package.py [hunk 4]
```diff
@@ -3,27 +3,32 @@
...
+    depends_on('binutils@2.26:', type='run')
+    depends_on('coreutils',      type='run')
+    depends_on('gmake',          type=('run', 'build'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/flit/package.py:30]: I've never seen gmake listed as a run dependency before. It's usually only needed at build time

File: var/spack/repos/builtin/packages/flit/package.py [hunk 5]
```diff
@@ -3,27 +3,32 @@
...
-    @property
-    def install_targets(self):
-        return ['install', 'PREFIX=%s' % self.prefix]
+    depends_on('python@3:',      type='run')
+    depends_on('py-toml',        type='run')
+    depends_on('py-pyelftools',  type='run')
+
+    #depends_on('bash',           type='run')
+    depends_on('binutils@2.26:', type='run')
+    depends_on('coreutils',      type='run')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/flit/package.py:29]: We don't normally add binutils/coreutils as package dependencies, see https://spack.readthedocs.io/en/latest/getting_started.html#binutils

File: var/spack/repos/builtin/packages/py-pyelftools/package.py [hunk 6]
```diff
@@ -10,6 +10,9 @@ class PyPyelftools(PythonPackage):
     """A pure-Python library for parsing and analyzing ELF files and DWARF
        debugging information"""
     homepage = "https://pypi.python.org/pypi/pyelftools"
-    url      = "https://pypi.io/packages/source/p/pyelftools/pyelftools-0.23.tar.gz"
+    url      = "https://pypi.io/packages/source/p/pyelftools/pyelftools-0.26.tar.gz"
 
+    version('0.26', sha256='86ac6cee19f6c945e8dedf78c6ee74f1112bd14da5a658d8c9d4103aed5756a2')
     version('0.23', sha256='fc57aadd096e8f9b9b03f1a9578f673ee645e1513a5ff0192ef439e77eab21de')
+
+    depends_on('py-setuptools', type='build')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pyelftools/package.py:18]:
```suggestion
    depends_on('py-setuptools', when='@0.25:', type='build')
```

File: var/spack/repos/builtin/packages/flit/package.py (hunk 1 above)
Comment (mikebentley15) [reply var/spack/repos/builtin/packages/flit/package.py]: Done

File: var/spack/repos/builtin/packages/flit/package.py (hunk 2 above)
Comment (mikebentley15) [reply var/spack/repos/builtin/packages/flit/package.py]: Done

File: var/spack/repos/builtin/packages/flit/package.py (hunk 3 above)
Comment (mikebentley15) [reply var/spack/repos/builtin/packages/flit/package.py]: Done

File: var/spack/repos/builtin/packages/flit/package.py (hunk 5 above)
Comment (mikebentley15) [reply var/spack/repos/builtin/packages/flit/package.py]: Done

File: var/spack/repos/builtin/packages/py-pyelftools/package.py (hunk 6 above)
Comment (mikebentley15) [reply var/spack/repos/builtin/packages/py-pyelftools/package.py]: Done

File: var/spack/repos/builtin/packages/flit/package.py (hunk 4 above)
Comment (mikebentley15) [reply var/spack/repos/builtin/packages/flit/package.py:27]: FLiT uses GNU make to function properly.  It is a true runtime dependency, more so than a build dependency.  FLiT generates GNU Makefiles to be used, similar to CMake.

**Comment by mikebentley15** [comment]
Thanks for completing the review so quickly!  And such good feedback!  Thank you

**Pull request merged**